tests/dogtag/pytest-ansible/provision/osp_provision.py

```diff
@@ -0,0 +1,176 @@
+import os
+import random
+import sys
+
+import click
+import openstack
+
+
+OS_AUTH_URL = os.getenv('OS_AUTH_URL')
+OS_PROJECT_NAME = os.getenv('OS_PROJECT_NAME')
+OS_USERNAME = os.getenv('OS_USERNAME')
+OS_REGION_NAME = os.getenv('OS_REGION_NAME')
+OS_KEYPAIR_NAME = os.getenv('OS_KEYPAIR_NAME')
+CI_FLAVOR_NAME = os.getenv('CI_FLAVOR_NAME')
+DOMAIN_NAME = os.getenv('DOMAIN_NAME')
+OS_PASSWORD = os.getenv('OS_PASSWORD')
+OS_NETWORK_ID = os.getenv('OS_NETWORK_ID').split(',')
+OS_NETWORK_ID = random.choice(OS_NETWORK_ID)
+
+inventory_file = "ansible_inventory"
+instance_record = '/tmp/instance_record.txt'
+
+
+def create_connection(auth_url, region, project_name, username, password):
+    return openstack.connect(auth_url=auth_url,
+                             project_name=project_name,
+                             username=username,
+                             password=password,
+                             user_domain_name=DOMAIN_NAME,
+                             project_domain_name=DOMAIN_NAME,
+                             region_name=region)
+
+
+def spawn_server(conn, image_name, server_type='master'):
+    print("Spawning server: ")
+
+    if server_type == 'master':
+        server_name = 'OSP_API_Master_{}'.format(random.randint(999, 99999))
+    else:
+        server_name = 'OSP_API_Client_{}'.format(random.randint(999, 99999))
+    image = conn.compute.find_image(image_name)
+    flavor = conn.compute.find_flavor(CI_FLAVOR_NAME)
+    try:
+        server = conn.compute.create_server(name=server_name, image_id=image.id,
+                                            flavor_id=flavor.id, networks=[{"uuid": OS_NETWORK_ID}],
+                                            key_name=OS_KEYPAIR_NAME)
+
+        server = conn.compute.wait_for_server(server, status='ACTIVE', failures='ERROR', interval=2, wait=420)
+        return server
+    except Exception as e:
+        print(e)
+        sys.exit(1)
+
+
+def delete_server(conn, server_name):
+    print("Removing Server")
+    server = conn.compute.find_server(server_name)
+    conn.compute.delete_server(server)
+
+
+def list_images(conn, name):
+    print("List Images:")
+    image_list = []
+    for image in conn.image.images():
+        if ("latest" in image.name and image.name.startswith(name)) or (name in image.name):
+            image_list.append(image.name)
+            print("Found Image: {}".format(image.name))
+    return image_list
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option('--image', default='rhel-8.1.0', help="Image to provision. (Ex. rhel-8.1.0)")
```

Comment:
I think the default is `Fedora 31` based on the actual code below.

Reply:
Thanks endi.! Done changes